source/common/runtime/runtime_impl.cc

```diff
@@ -201,17 +227,47 @@ SnapshotImpl::SnapshotImpl(RandomGenerator& generator, RuntimeStats& stats,
   stats.num_keys_.set(values_.size());
 }
 
-Snapshot::Entry SnapshotImpl::createEntry(const std::string& value) {
-  Entry entry{value, absl::nullopt};
-  // As a perf optimization, attempt to convert the entry's string into an integer. If we don't
-  // succeed that's fine.
-  uint64_t converted;
-  if (StringUtil::atoul(entry.string_value_.c_str(), converted)) {
-    entry.uint_value_ = converted;
-  }
+SnapshotImpl::Entry SnapshotImpl::createEntry(const std::string& value) {
+  Entry entry;
+  entry.entry_type_ = Entry::EntryType::STRING_VALUE;
+  entry.raw_string_value_ = value;
+
+  // As a perf optimization, attempt to parse the entry's string and store it inside the struct. If
+  // we don't succeed that's fine.
+  resolveEntryType(entry);
+
   return entry;
 }
 
+bool SnapshotImpl::parseEntryUintValue(Entry& entry) {
+  // TODO (tonya11en): Use C++17 if-statements with initializers to limit scope of variables.
```

Comment:
nit: This applies all over the place and I'm not sure it is worth putting a TODO for. I would remove.